src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceOpProcessorImpl.scala

```diff
@@ -69,7 +75,8 @@ private[tracker] object InstanceOpProcessorImpl {
     private[this] def expungeInstance(id: Instance.Id)(implicit ec: ExecutionContext): Future[InstanceUpdateEffect] = {
       directInstanceTracker.instance(id).map {
         case Some(existingInstance: Instance) =>
-          InstanceUpdateEffect.Expunge(existingInstance)
+          val events = eventsGenerator.events(InstanceStatus.Killed, existingInstance, task = None, clock.now())
```

Comment:
What about Terminating (read: this is a termination in progress). Killing might be an instance of Terminating. Suicidal might be an instance of Terminating (meaning we're expecting that Mesos will kill any remaining tasks)